mono/mini/debugger-agent.c

```diff
@@ -1755,6 +1759,7 @@ start_debugger_thread (MonoError *error)
 	debugger_thread_handle = mono_threads_open_thread_handle (thread->handle);
 	g_assert (debugger_thread_handle);
 	
+	debugger_thread_exited = FALSE;
```

Comment:
Does this value get checked anywhere?

Reply:
Yes it's used here: https://github.com/Unity-Technologies/mono/blob/unity-main/mono/mini/debugger-agent.c#L1707 -- the original implementation that used this never reset the value when the debugger thread got started. I don't think the init -> stop -> restart use case was considered at that time.